aiohttp/web_protocol.py

```diff
@@ -449,17 +449,19 @@ def _process_keepalive(self) -> None:
             return
 
         next = self._keepalive_time + self._keepalive_timeout
+        now = self._loop.time()
 
         # handler in idle state
-        if self._waiter:
-            if self._loop.time() > next:
-                self.force_close()
-                return
+        if self._waiter and now > next:
+            self.force_close()
+            return
 
         # not all request handlers are done,
-        # reschedule itself to next second
-        self._keepalive_handle = self._loop.call_later(
-            self.KEEPALIVE_RESCHEDULE_DELAY,
```

Comment:
Ironically, this was introduced to fix a CPU spike: #2701.

I wonder if we can improve this further and avoid the 1 second reschedules entirely? From what I can tell, the 1 second rechecks will be happening while a request is being processed (i.e. we are not waiting for a new message, so there is no _waiter).

Can't we just reschedule once the request handler has finished processing? It appears to me the waiter is only set in start(), and there's no way _keepalive_time won't get updated just before creating another waiter:
https://github.com/aio-libs/aiohttp/blob/4d604eafefbee22ad35886c3bd12e4c3048cc45b/aiohttp/web_protocol.py#L612

Therefore, it seems to me that if we reach this point and `now > next`, then there's no point in rescheduling here, we can just exit and let it schedule again after updating the timeout.

Reply:
I'll adjust this a bit to reschedule if it fires too early and avoid the 1s reschedules.